src/libsync/account.cpp

```diff
@@ -85,18 +85,7 @@ Account::~Account() = default;
 
 QString Account::davPath() const
 {
-    if (capabilities().chunkingNg()) {
-        // The chunking-ng means the server prefer to use the new webdav URL
-        return QLatin1String("/remote.php/dav/files/") + davUser() + QLatin1Char('/');
-    }
-
-    // make sure to have a trailing slash
-    if (!_davPath.endsWith('/')) {
-        QString dp(_davPath);
-        dp.append('/');
-        return dp;
-    }
-    return _davPath;
+    return QLatin1String("/remote.php/dav/files/") + davUser() + QLatin1Char('/');
```

Comment:
I think this changes the behaviour. Should we not construct the path from `_davPath` ?

Reply:
I guess that is intentional given the commit message that introduced that code
```
libsync: use the new webdav url if the server reports it

The rules to select the webdav url are now:

 - If the server reports that the new chunking algorithm is working,
   always use remote.php/dav/files/<username>
   This capability can be overriden with an environment variable

 - Otherwise, use the dav path provided by the theme, which defaults to
   remote.php/webdav

This means that with the newer server, the branding can no longer override
the webdav URL. If there is still an usecase for the branding to do so, we
need to find another way to override it. But it is now more complicated to
configure as might need include the username and need different endpoint
depending on the operations (chunks or not)

Issue #4007
```

Reply:
@mgallien But, `_davPath` is always initialized with `QString Theme::webDavPath()` in the `Account`'s constructor. So, are you still sure this is the correct fix? I am not sure I fully understand it.

Reply:
At least if we decide to hardcode the path, we should remove `Theme::webdav()` otherwise the code will be confusing, and we should also remove the `_davPath` member variable. It's then not needed anymore. `setNonShib()` should also be obsolete as we have removed the Shibbolteh code.